Adding Explanation of 223P

lazlop opened the pull request:
Close #21.

@steveraysteveray I added some documentation introducing 223P and explaining basically what information is modeled with 223P. If you have time, can you review it? 

I'm not sure if the explanation is comprehensive and covers all the ideas we should introduce, and I'm also not sure if I get too far into the weeds too fast. I still have a few to-dos, like moving my image out of _static to the proper image directory and linking other sections that have yet to be written, but it would be good to get your thoughts on the text. Let me know what you think!

lazlop commented:
@gtfierro @MatthewSteen 
If either of you get the chance, please give my first draft of the text a review and let me know what you think!

gtfierro commented on explanation/what-is-modeled.md:
```diff
@@ -0,0 +1,41 @@
+
+## Type
+
+This standard provides well defined classes used to describe entities relevant to building system information. Entities in s223 models will generally be instances of the many classes defined or referred to by the standard. The classes in the standard provide 'names' for the fundamental building blocks used in 223P models (e.g. a fan will be an instance of the class [s223:Fan](https://explore.open ...
```
Use "223" instead of "s223" or "223P"

gtfierro commented on explanation/what-is-modeled.md:
```diff
@@ -0,0 +1,41 @@
+
+## Topology
+
+This standard can be used to describe the topology of the equipment and spaces in a building, but not the geometric details. Topology refers to the way entities are connected and how some media (e.g. water, air, or electricity) is conveyed between them. There are several different classes used to describe which entities participate in connections and how they connect. This describe [Connectables] ...
```
ConnectionPoints definition change to "which model where Connectables can be connected"

gtfierro commented on explanation/what-is-modeled.md:
```diff
@@ -0,0 +1,41 @@
+
+This standard can be used to describe the topology of the equipment and spaces in a building, but not the geometric details. Topology refers to the way entities are connected and how some media (e.g. water, air, or electricity) is conveyed between them. There are several different classes used to describe which entities participate in connections and how they connect. This describe [Connectables] ...
+
+(not sure of we should include this) There are also multiple relation used to describe the details of these connections, and how the multiple entities involved in a connection relate to each other. Figure 2 summarizes these relations (TODO maybe add figure with all the connections relationships). Additionally, there is a long list of Mediums to define the substances that are conveyed (e.g. gas, e ...
```
Where is Figure 2? You should put it here, refer to it as demonstrating what relationships exist between Connectables, ConnectionPoints, and Connections. You should also refer to the `model_inference` page as an aside to let readers know that when they are creating a model, they don't need to remember all of these. Just use `cnx`

gtfierro commented on explanation/definitions.md:
`````diff
@@ -0,0 +1,70 @@
+````
+The 223P standard defines a hierarchy of classes used to define the entities within a buidling. This section provides a basic definitino of the classes at the top level of the hierarchy to help users understand what the standard aims to represent, which is described in the next section (link). 
+
+ - **Connectable:** An abstract class representing a thing that can be connected via ConnectionPoints and Connections.
`````
Mention that this is where Equipment can be found, among other things

gtfierro replied:
Looks like you did this below. Remove the duplicate Connectable entry here

gtfierro commented on explanation/definitions.md:
`````diff
@@ -0,0 +1,70 @@
+## 223P Top Level Classes
+````{margin}
+```{note}
+A little referency, but it's hard to understand any of the explanation without these things.
`````
I think this is fine to exist here

gtfierro commented on explanation/what-is-modeled.md:
```diff
@@ -0,0 +1,41 @@
+
+(TODO: If there is a good example in Guides, this should link to it)
+
+## Composition
```
100% use Composition here, not Containment

gtfierro commented on explanation/definitions.md:
```diff
@@ -0,0 +1,70 @@
+Examples of relationships are *encapsulation* (one entity is contained within another), *sequence* (one entity takes effect before another in some process) and *instantiation* (one entity's type is given by another entity).
+
+- **Relation:** A predicate (RDF property) used to describe a given relationship.
+Examples of a relation are the s223 relation s223:contains, which relationship between two pieces of equipment in which one contains another.  
```
Put "s223:contains" in backticks, e.g. `s223:contains`

gtfierro replied:
Carry this through everywhere

gtfierro commented on explanation/definitions.md:
`````diff
@@ -0,0 +1,70 @@
+Borrowed from brick. I remember finding this very helpful.
+```
+````
+ - **Entity:** An entity is an abstraction of any physical, logical or virtual item; the actual "things" in a building. Examples are mechanical equipment such as air handling units, luminaires, spatial elements like rooms, or the area of these rooms served by certain mechanical equipment. Virtual and logical entities may represent the various data points in a BAS system, such as sensing and actua ...
`````
Virtual and logical entities are not represented in 223, so I would remove this part

lazlop replied:
Just remove virtual and logical parts, right?
I definitely agree with this change, as using entity to refer to "things that can have properties" in the standard is much more useful than including properties and functionblocks in the definition of entities ( I think those are our representation of virtual and logical elements).

gtfierro replied:
Yeah, that should do it

MatthewSteen reviewed (commented):
Some high level comments based on what I think the goal of the Explanation section should be.
1. Introduce the Standard (1st file).
2. Provide a high level summary of it (2nd file).

# 223 Introduction (223p.md)
1. Rename this file to `223_introduction.md` or similar (same as heading 1 above)?
3. Write a brief opening sentence...
   > The Purpose, Scope, and Definitions from the Standard are copied below as an introduction to understanding it.

## Purpose
I think we should move the `Purpose` from `intro.md` here. 

## Scope
I think we should move the `Scope` from `intro.md`.

## Definitions
I think we should move the `Definitions` from `definitions.md` here and delete that file.

# 223 Overview (what-is-modeled.md)
1. Rename to `223_overview.md` or similar (same as heading 1 above). 
2. Summarize Section 4 of the Standard?
4. Summarize Section 5-14 of the Standard?

MatthewSteen commented:
> @MatthewSteen I think adding all of the definitions to the same page as the scope and intro makes the page a little bit long. What if we keep the definitions file and just make it the next page after the intro in the table of contents?

I don't think it does. I also don't think we need to re-write what's already in the Standard for these pages. Simply copy/paste the Purpose, Scope, and Definitions into the 1st Explanation page. The Definitions section of the Standard is < 1 page in the docx.

> Also do we need more info on the landing page? or is it ok if it's a little empty.

I think it should be short. 
1. What this is.
2. Title of the Standard with disclaimers, notes, etc.
3. Autogenerated TOC.

MatthewSteen commented:
@lazlop here's what I'm thinking/it would look like...

# 223 Introduction
The Purpose, Scope, and Definitions from the Standard are copied below as an introduction to understanding it.

## Purpose
>The purpose of this standard is to define formal knowledge concepts and a methodology to apply them to create interoperable, machine-readable semantic frameworks for representing building automation and control data, and other building system information

## Scope
>This standard provides a comprehensive way to apply semantic formalisms to represent the context of building system data and relationships between the associated building mechanical system components so that software applications can find and understand the information in an automated way. It is intended to facilitate the development and implementation of building analytics tools and enterprise knowledge applications that can implement many building system functions, including:
>1. automated fault detection and diagnostics,
>2. building system commissioning,
>3. digital twins
>4. optimization of energy use, and
>5. smart grid interactions.

## Definitions
>Connectable: an abstract class that represents a thing (Equipment or DomainSpace) that can be connected via ConnectionPoints and Connections.
Connection: the modeling construct used to represent a physical thing (e.g., pipe, duct, or wire) that is used to convey some Medium (e.g., water, air, or electricity) between two Connectable things.
ConnectionPoint: an abstract modeling construct used to represent the fact that one Connectable thing can be connected to another Connectable thing using a Connection. It is the abstract representation of the flange, wire terminal, or other physical feature where a connection is made.
Domain: a categorization of building service or specialization used to characterize equipment or spaces in a building. Example domains include HAVC, lighting, and plumbing.
DomainSpace: a portion or the entirety of a PhysicalSpace that is associated with a Domain, such as lighting, HVAC, or physical security. DomainSpaces can be combined to form a Zone.
Duct: a subclass of Connection that represents a conduit through which air is conveyed.
ElectricWire: a subclass of Connection that represents one or more electrical conductors used to convey electricity.
Equipment: the modeling construct used to represent a mechanical device designed to accomplish a specific task that one might buy from a vendor. Examples include a pump, fan, heat exchanger, luminaire, temperature sensor, or flow meter. A piece of equipment can contain another piece of equipment. For example, an air handling unit can contain a cooling coil.
PhysicalSpace: an architectural concept that can represent a room, a collection of rooms such as a floor, a part of a room, or any physical space that might not even be thought of as a room, such as a patio.
Pipe: a subclass of Connection that represents a hollow cylinder of metal or other material used to convey a Medium.
System: a task-oriented collection of interacting or interrelated Equipment defined by the modeler. Examples of possible systems are an air distribution system, or a hot water system. Systems can contain other Systems.
Zone: a collection of DomainSpaces of a specific domain that are grouped together from the perspective of building services or controls.

lazlop commented:
Those definitions are a little out of date, so I think we should update the publication document to what I've added here. 
I will create in issue on the s223 gitlab to remind us of that.

MatthewSteen commented:
Sounds good. Do you think we need a second file to go into more detail? We could just stop at an `223_introduction.md` since presumably the ontology will be public and the additional detail will be there. 

Once it is publicly hosted we could also automate updating these sections from it.

lazlop commented:
I don't think we need any more detail than what is on the current definitions file, which just adds some terms needed to understand the introduction and updates definitions for stuff in the standard

lazlop commented:
@gtfierro @MatthewSteen 
Ok I've adjusted things based on your feedback.

After merging in main I am no longer able to build the jupyter book locally though. Are have the commands needed for that changed

MatthewSteen commented:
Probably because there are dependencies for executing the code now. I've started https://github.com/open223/docs.open223.info/wiki/Developer-Documentation to help with that.

lazlop reviewed (commented):
changes made

lazlop commented:
@MatthewSteen 

I'm not sure I like this flow as much, but please review and let me know what you think!

gtfierro commented:
I think the flow is ok but maybe "223 Introduction" wants to be "Purpose and Scope"? Then we can move Definitions and Top Level Classes to a new page ("Important(?) Definitions and Concepts")

There's a broken link for "223 Modeling Example" and "Relations expressing different perspectives of connection" (fix spelling on "perspectives")

I

lazlop commented:
Yeah, inserting images the html way doesn't work unless the images are in the static folder at the repo level. I fixed the broken image links. 
I updated to what you suggested Gabe. This is pretty much where we started but with the "Using Graph Concepts" section on the overview page.

lazlop commented:
@MatthewSteen @gtfierro 
I think I prefer the original layout with Purpose and Scope on the landing page, but this also works for me if it works for you

MatthewSteen commented on intro.md:
```diff
@@ -6,32 +7,17 @@ The purpose of this site is to provide documentation for users of the *proposed*
-The Purpose and Scope from from the current standard are provided below to introduce it to users. For more information about the standard, please see the published document [^1].
-
-[^1]: TODO link to public review.
+Standard 223 defines modeling constructs for use in creating a machine-readable representation of building systems, the building spaces that they serve, and the measurement and control points used to provide a safe and comfortable environment for the building occupants. The standard can be considered a toolkit of components and rules for using them to create a semantic model of a particular build ...
```
```suggestion
```

Remove as this is copied from Section 4 and should come later.

MatthewSteen commented on explanation/223-introduction.md:
```diff
@@ -0,0 +1,18 @@
+[^1]: TODO link to public review.
+
+## 1. Purpose
+The purpose of this standard is to define formal knowledge concepts and a methodology to apply them to create interoperable, machine-readable semantic frameworks for representing building automation and control data, and other building system information.
```
```suggestion
>The purpose of this standard is to define formal knowledge concepts and a methodology to apply them to create interoperable, machine-readable semantic frameworks for representing building automation and control data, and other building system information.
```

MatthewSteen commented on explanation/223-introduction.md:
```diff
@@ -0,0 +1,18 @@
+The purpose of this standard is to define formal knowledge concepts and a methodology to apply them to create interoperable, machine-readable semantic frameworks for representing building automation and control data, and other building system information.
+
+## 2. Scope
+This standard provides a comprehensive way to apply semantic formalisms to represent the context of building system data and relationships between the associated building mechanical system components so that software applications can find and understand the information in an automated way. It is intended to facilitate the development and implementation of building analytics tools and enterprise k ...
+1. automated fault detection and diagnostics
+2. building system commissioning
+3. digital twins
+4. real time optimization
+5. smart grid interactions
```
```suggestion
>This standard provides a comprehensive way to apply semantic formalisms to represent the context of building system data and relationships between the associated building mechanical system components so that software applications can find and understand the information in an automated way. It is intended to facilitate the development and implementation of building analytics tools and enterprise knowledge applications that can implement many building system functions, including:
>1. automated fault detection and diagnostics,
>2. building system commissioning,
>3. digital twins
>4. optimization of energy use, and
>5. smart grid interactions.
```

MatthewSteen reviewed (commented):
Rename file to `purpose_scope.md` (snake case convention per Python).

MatthewSteen commented:
For the 2nd explanation, I think it should be named `definitions_concepts.md` with the following content.

# Definitions and Concepts
[intro of some sort]

## Definitions (quoted directly from current GitLAB artifact)
>Connectable: an abstract class that represents a thing (Equipment or DomainSpace) that can be connected via ConnectionPoints and Connections.
Connection: the modeling construct used to represent a physical thing (e.g., pipe, duct, or wire) that is used to convey some Medium (e.g., water, air, or electricity) between two Connectable things.
ConnectionPoint: an abstract modeling construct used to represent the fact that one Connectable thing can be connected to another Connectable thing using a Connection. It is the abstract representation of the flange, wire terminal, or other physical feature where a connection is made.
Domain: a categorization of building service or specialization used to characterize equipment or spaces in a building. Example domains include HAVC, lighting, and plumbing.
DomainSpace: a portion or the entirety of a PhysicalSpace that is associated with a Domain, such as lighting, HVAC, or physical security. DomainSpaces can be combined to form a Zone.
Duct: a subclass of Connection that represents a conduit through which air is conveyed.
ElectricWire: a subclass of Connection that represents one or more electrical conductors used to convey electricity.
Equipment: the modeling construct used to represent a mechanical device designed to accomplish a specific task that one might buy from a vendor. Examples include a pump, fan, heat exchanger, luminaire, temperature sensor, or flow meter. A piece of equipment can contain another piece of equipment. For example, an air handling unit can contain a cooling coil.
PhysicalSpace: an architectural concept that can represent a room, a collection of rooms such as a floor, a part of a room, or any physical space that might not even be thought of as a room, such as a patio.
Pipe: a subclass of Connection that represents a hollow cylinder of metal or other material used to convey a Medium.
System: a task-oriented collection of interacting or interrelated Equipment defined by the modeler. Examples of possible systems are an air distribution system, or a hot water system. Systems can contain other Systems.
Zone: a collection of DomainSpaces of a specific domain that are grouped together from the perspective of building services or controls.

## Concepts
[beginning of "original" content that isn't quoted from standard]

MatthewSteen commented:
The 3rd explanation can be a *brief* description of the underlying technologies from the `graph-structure.md` *and* the https://github.com/open223/docs.open223.info/blob/main/reference/other-resources.md#semantic-web-technologies section.

lazlop commented:
I've updated for some of these changes. 
I think I prefer having 1) scope and purpose 2) definitions 3) graph structure (definitions and graph structure can be on the same page because they serve a similar role) and 4) 223 overview. Reading through linearly I think that provides information in the best order to build up to the 223 overview, which relies on the graph structure and definitions explanations. 

I have changed it to what you suggested. Lets let this simmer (or merge it) to see if my preference is just based on familiarity and to see what others think.

MatthewSteen commented:
Ok, I'll approve, you can merge, and we can revisit later.

steveraysteveray commented:
Wow, just catching up with all this activity. 

In 223_overview.md, 

1. One minor comment/question: Do you think the heading "Type" might confuse some, especially CS people, versus calling it "Class"? 
2. Figure 1 doesn't look right, but maybe that's because I'm looking at the page from the web interface?
3. constructs include include [Equipment](https://explore.open223.info/s223/Equipment.html). Delete repeated word.
4. "For example, an auditorium (a PhysicalSpace) may enclose several different areas served by independently controlled lights (DomainSpaces)." ...add to read: "For example, an auditorium (a PhysicalSpace) may enclose several different areas served by independently controlled lights (DomainSpaces), as well as enclosing DomainSpaces for different domains (e.g. heating, lighting...).
5. "Properties also have many different characteristics which are described by a vocabulary of [EnumerationKinds](https://explore.open223.info/s223/EnumerationKind.html)." ...edit to "characteristics (called Aspects)..."
